Dockerfile

```diff
@@ -20,30 +20,41 @@ RUN git submodule update --init --recursive \
     && rm -rf `find . -type d -name ".git"` \
     && rm .gitmodules
 
-FROM alpine:3.12 AS organizer
+FROM alpine:3.12 AS builder
 
-# Files for the build process
-WORKDIR /out/build
+# Install cmake
+RUN apk --no-cache add cmake clang clang-dev make gcc g++ libc-dev linux-headers
+
+# Build
 COPY include include
 COPY lib lib
 COPY src src
 COPY thirdparty thirdparty
 COPY CMakeLists.txt .
-COPY --from=submodule-initializor /out /out/build
+COPY --from=submodule-initializor /out .
+RUN cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=Release -DARK_BUILD_EXE=1 \
+    && cmake --build build \
+    && rm -rf lib/ext
+
+FROM alpine:3.12 AS organizer
+
+# Files needed to run Ark
+WORKDIR /out/ark
+COPY --from=builder build build
+COPY --from=builder submodules submodules
+COPY --from=builder lib lib
 
 # The docker-entrypoint.sh file
 COPY --from=permissions-giver /out/docker-entrypoint.sh /out/docker/docker-entrypoint.sh
 
-FROM alpine:3.12 AS builder
+FROM alpine:3.12 AS runner
 
 # Install cmake
 RUN apk --no-cache add cmake clang clang-dev make gcc g++ libc-dev linux-headers
```

Comment:
I'm not sure but I think we just need cmake and maybe make to have this working, the whole compiler thing could maybe be avoided

Reply:
You're correct!